src/System.Text.Json/ref/Configurations.props

```diff
@@ -3,6 +3,7 @@
     <PackageConfigurations>
       netstandard;
       net461;
+      netcoreapp3.0;
```

Comment:
you want to add a netcoreapp3.0 asset to the System.Text.Json package?

Reply:
yes I will add it to 3.1 branch as well. cc @ericstj can explain the reason better than me

Reply:
Without this, we bring in the `netstandard2.0` ref on `netcoreapp3.0` which in-turn brings in the `Microsoft.BCL.AsyncInterfaces` package.  We'd like to avoid bringing in this dependency on `netcoreapp3.0`.  /cc @ahsonkhan @joperezr

Reply:
An alternative here could be to target `netstandard2.1` to avoid the dependency.  With that we wouldn't need BuildConfigurations, I think.

Reply:
I see I suppose this is because conflict resolution will find that netstandard2.0 asset has a higher assembly version than inbox? I also guess that we don't want to put a placeholder on netcoreapp3.0 TFM in order to not freeze the API?

Reply:
Would this cause a problem with roll-forward?

Reply:
> Would this cause a problem with roll-forward?

Note that we're only changing the ref here, which is not subject to roll-forward.  The src lib already includes the nca3.0 config.

The goal with System.Text.Json is to simultaneously rev package and inbox, similar to S.C.Immutable/Reflection.Metadata.  The reason for the nca3.0 ref assembly is so that we accurately represent the distinct configs of the library.  We were already including the 3.0 lib in the package.  Omitting the 3.0 ref *works* (since there is no surface area unique to nca3.0) but its less than ideal.  Including the 3.0 ref lets us avoid depending on the compat packages just for compile.  We could choose to do ns2.1 instead, but I think it makes more sense to keep parallel configs to the src lib.

Reply:
Actually took a closer look here and I think we should just clean this up.  The reference isn't even shipping inside the package, so my concern about leaking the Microsoft.BCL.* package to netcoreapp3.0 is a non-issue.  Instead of doing this, can we delete the PackageConfigurations (just use BuildConfigurations) and remove this ProjectReference from the pkgproj?  Not sure if we had a reason for doing it the way it is now, or we just missed cleaning this up before.  @joperezr do you recall?